lib/Driver/Driver.cpp

```diff
@@ -342,6 +344,13 @@ static bool populateOutOfDateMap(InputInfoMap &map, StringRef argsHashStr,
   }
 
   if (!optionsMatch) {
+    if (ShowIncrementalBuildDecisions) {
+      llvm::outs() << "Incremental compilation has been disabled, because "
+                   << "different arguments were passed to the compiler.\n"
+                   << "\tCompiling with argument hash: " << argsHashStr << "\n"
```

Comment:
I guess we could just store the old argument list verbatim, in addition to the hash? The only time you'll see it is when you change something that does affect the output.